packages/lib/shared/utils/tokenDisplay.ts

```diff
@@ -0,0 +1,36 @@
+import { ZERO_VALUE_DASH, Numberish, isZero } from './numbers'
+
+/**
+ * Formats any value for display, showing dash for falsy values (0, '0', '', undefined)
+ * @param value - The value to format (string, number, or undefined)
+ * @param formatter - Optional formatter function for non-falsy values
+ * @param options - Options to pass to the formatter function
+ * @returns Formatted display string or dash for falsy values
+ */
+export function formatFalsyValueAsDash(
+  value: Numberish | undefined,
+  formatter?: (value: Numberish, options?: any) => string,
+  options?: any
```

Comment:
can you help me understand the need to bring `formatter` and `options` into this function? 

is it not possible to keep formatting as it was and then check if the post-formatted value is falsy?

Reply:
@MattPereira 

checking the value post-formatted might be more complex
and the logic (is value zero or undefined and should it be shown as a dash) is pretty easy but repetitive so as an example i quickly created a small func that will check for falsey and if not will format the value accordingly
i see it is pretty much copied unchanged
i took another look and it seems to be working ok?